client/src/components/InputsModal/InputHelpers.ts

```diff
@@ -0,0 +1,156 @@
+        const authType = (authJson.auth_type ||
+          input.options.components.find((c) => c.name === 'auth_type')?.default) as AuthType;
+
+        // Determine which fields are required; the `authValues` and `components` props
+        // in getAuthFields() and getAccessFields() are irrelevant for this
+        const fields =
+          input.options?.mode === 'auth'
+            ? getAuthFields(authType, new Map(), [])
+            : getAccessFields(authType, new Map(), []);
+        const requiredFields = fields.filter((field) => !field.optional).map((field) => field.name);
+        authMissingRequiredInput = requiredFields.some((field) => !authJson[field as keyof Auth]);
+      } catch (e: unknown) {
+        const errorMessage = e instanceof Error ? e.message : String(e);
+        enqueueSnackbar(`Auth info inputs incorrectly formatted: ${errorMessage}`, {
+          variant: 'error',
+        });
+        return true;
+      }
+    }
+
+    // If input has OAuth, check if required values are filled
+    let oAuthMissingRequiredInput = false;
+    if (input.type === 'oauth_credentials') {
+      try {
+        const oAuthJson = JSON.parse(
+          (inputValue as string) || '{ "access_token": null }',
+        ) as OAuthCredentials;
+        const accessTokenIsEmpty = !oAuthJson.access_token;
+        const refreshTokenIsEmpty =
+          !!oAuthJson.refresh_token && (!oAuthJson.token_url || !oAuthJson.client_id);
+        oAuthMissingRequiredInput = (!input.optional && accessTokenIsEmpty) || refreshTokenIsEmpty;
+      } catch (e: unknown) {
+        const errorMessage = e instanceof Error ? e.message : String(e);
+        enqueueSnackbar(`OAuth credentials incorrectly formatted: ${errorMessage}`, {
+          variant: 'error',
+        });
+        return true;
+      }
+    }
+
+    return (
+      (!input.optional && !inputValue) || oAuthMissingRequiredInput || authMissingRequiredInput
+    );
+  });
+};
+
+export const parseSerialChanges = (
+  inputType: string,
+  changes: string,
+  setInvalidInput: (isInvalidInput: boolean) => void,
+): TestInput[] | undefined => {
+  let parsed: TestInput[];
+  try {
+    parsed = (inputType === 'JSON' ? JSON.parse(changes) : YAML.load(changes)) as TestInput[];
+    // Convert OAuth/Auth input values to strings; parsed needs to be an array
+    parsed.forEach((input) => {
+      if (input.type === 'oauth_credentials' || input.type === 'auth_info') {
+        input.value = JSON.stringify(input.value);
+      }
+    });
+    setInvalidInput(false);
+    return parsed;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  } catch (e) {
+    // Ignore errors; assume user error
+    setInvalidInput(true);
+    return undefined;
+  }
+};
+
+export const serializeMap = (
+  inputType: string,
+  inputs: TestInput[],
+  map: Map<string, unknown>,
+): string => {
+  const flatObj = inputs.map((requirement: TestInput) => {
+    // Parse out \n chars from descriptions
```

Comment:
Why are newlines being removed?

Reply:
For display purposes, the new lines and the modal spacing together look really bad. Descriptions shouldn’t actually be changed. I’m not sure what the interaction is with the markdown descriptions.

Reply:
Supporting newlines was one of the immediate drivers to use markdown in descriptions, so I"m not sure how to reconcile that, but it can be handled in that ticket.

Reply:
Yeah this is preexisting behavior so probably shouldn’t be fixed here